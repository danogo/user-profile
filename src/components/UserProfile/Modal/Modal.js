import React from 'react';
import styles from './Modal.module.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const Modal = (props) => {
  const { onBtnClick, showModal  } = props;
  const modalClassName = showModal ? styles.showModal : styles.hideModal;

  return (
    <div className={modalClassName}>
      <div className={styles.modalBox}>
        <input id='inputModal' value={window.location.href} onFocus={e => e.target.select()} readOnly/>
        <button onClick={onBtnClick}>
          <FontAwesomeIcon icon={['fas', 'times']} color='#FFFFFF'/>
        </button>
      </div>
    </div>
  );
} 

export default Modal;