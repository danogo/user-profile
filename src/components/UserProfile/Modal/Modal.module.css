.showModal {
  display: block;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.6);
}                 

.hideModal {
  display: none;
}

.modalBox {
  position: absolute;
  background: white;
  width: 80%;
  height: auto;
  top: 50%;
  left: 50%;
  transform: translate(-50%,-50%);
  max-width: 50rem;
  padding: 1.4rem 1.6rem;
  border-radius: 5px;
  box-shadow: 0 0 10px 2px #FFFFFF50;
}

.modalBox input {
  font-size: 1.6rem;
  border: none;
  padding: .6rem .8rem;
  background-color: #FFA640;
  color: #FFFFFF;
  border-radius: 5px;
  box-shadow: inset 0 0 10px rgba(0, 0, 0, 0.2);
  font-weight: bold;
}

.modalBox button {
  font-size: 1.4rem;
  color: #FFFFFF;
  letter-spacing: 1.2px;
  padding: .6rem .8rem;
  border-radius: 100%;
  background-color: #FFA640;
  border: none;
  cursor: pointer;
  font-weight: bold;
  float: right;
}

.modalBox button:hover {
  background-color: #FFBC70;
}