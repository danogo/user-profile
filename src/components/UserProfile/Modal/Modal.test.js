import React from 'react';
import { shallow } from 'enzyme';
import Modal from './Modal';

describe('Modal', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      onBtnClick: jest.fn(),
      showModal: false
    }
    wrapper = shallow(<Modal {...mockProps}/>);
  });

  test('component renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('modal is hidden by default', () => {
    expect(wrapper.hasClass('hideModal')).toBe(true);
  })

  test('receives onClick handler', () => {
    expect(wrapper.find('button').props().onClick).toBe(mockProps.onBtnClick);
  });

  test('callback is called when user clicks exit btn', () => {
    const btn = wrapper.find('button');
    btn.prop('onClick')();
    expect(mockProps.onBtnClick).toHaveBeenCalledTimes(1);
  })
});