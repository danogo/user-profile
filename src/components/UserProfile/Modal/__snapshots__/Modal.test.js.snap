// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    onBtnClick={[MockFunction]}
    showModal={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="modalBox"
      >
        <input
          id="inputModal"
          onFocus={[Function]}
          readOnly={true}
          value="http://localhost/"
        />
        <button
          onClick={[MockFunction]}
        >
          <FontAwesomeIcon
            border={false}
            className=""
            color="#FFFFFF"
            fixedWidth={false}
            flip={null}
            icon={
              Array [
                "fas",
                "times",
              ]
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
        </button>
      </div>,
      "className": "hideModal",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            id="inputModal"
            onFocus={[Function]}
            readOnly={true}
            value="http://localhost/"
          />,
          <button
            onClick={[MockFunction]}
          >
            <FontAwesomeIcon
              border={false}
              className=""
              color="#FFFFFF"
              fixedWidth={false}
              flip={null}
              icon={
                Array [
                  "fas",
                  "times",
                ]
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </button>,
        ],
        "className": "modalBox",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "inputModal",
            "onFocus": [Function],
            "readOnly": true,
            "value": "http://localhost/",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FontAwesomeIcon
              border={false}
              className=""
              color="#FFFFFF"
              fixedWidth={false}
              flip={null}
              icon={
                Array [
                  "fas",
                  "times",
                ]
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />,
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "border": false,
              "className": "",
              "color": "#FFFFFF",
              "fixedWidth": false,
              "flip": null,
              "icon": Array [
                "fas",
                "times",
              ],
              "inverse": false,
              "listItem": false,
              "mask": null,
              "pull": null,
              "pulse": false,
              "rotation": null,
              "size": null,
              "spin": false,
              "symbol": false,
              "title": "",
              "transform": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="modalBox"
        >
          <input
            id="inputModal"
            onFocus={[Function]}
            readOnly={true}
            value="http://localhost/"
          />
          <button
            onClick={[MockFunction]}
          >
            <FontAwesomeIcon
              border={false}
              className=""
              color="#FFFFFF"
              fixedWidth={false}
              flip={null}
              icon={
                Array [
                  "fas",
                  "times",
                ]
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </button>
        </div>,
        "className": "hideModal",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              id="inputModal"
              onFocus={[Function]}
              readOnly={true}
              value="http://localhost/"
            />,
            <button
              onClick={[MockFunction]}
            >
              <FontAwesomeIcon
                border={false}
                className=""
                color="#FFFFFF"
                fixedWidth={false}
                flip={null}
                icon={
                  Array [
                    "fas",
                    "times",
                  ]
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </button>,
          ],
          "className": "modalBox",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "inputModal",
              "onFocus": [Function],
              "readOnly": true,
              "value": "http://localhost/",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FontAwesomeIcon
                border={false}
                className=""
                color="#FFFFFF"
                fixedWidth={false}
                flip={null}
                icon={
                  Array [
                    "fas",
                    "times",
                  ]
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />,
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "",
                "color": "#FFFFFF",
                "fixedWidth": false,
                "flip": null,
                "icon": Array [
                  "fas",
                  "times",
                ],
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": null,
                "spin": false,
                "symbol": false,
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
